pkg/network/http/monitor_test.go

```diff
@@ -26,12 +31,209 @@ import (
 	"github.com/DataDog/datadog-agent/pkg/util/kernel"
 )
 
-func TestHTTPMonitorIntegration(t *testing.T) {
+const (
+	kb = 1024
+	mb = 1024 * kb
+)
+
+var (
+	emptyBody = []byte(nil)
+)
+
+func skipTestIfKernelNotSupported(t *testing.T) {
 	currKernelVersion, err := kernel.HostVersion()
 	require.NoError(t, err)
 	if currKernelVersion < kernel.VersionCode(4, 1, 0) {
 		t.Skip("HTTP feature not available on pre 4.1.0 kernels")
 	}
+}
+
+// TestHTTPMonitorLoadWithIncompleteBuffers sends thousands of requests without getting responses for them, in parallel
+// we send another request. We expect to capture the another request but not the incomplete requests.
+func TestHTTPMonitorLoadWithIncompleteBuffers(t *testing.T) {
+	skipTestIfKernelNotSupported(t)
+	slowServerAddr := "localhost:8080"
+	fastServerAddr := "localhost:8081"
+
+	slowSrvDoneFn := testutil.HTTPServer(slowServerAddr, testutil.Options{
+		SlowResponse: time.Millisecond * 500, // Half a second.
+		WriteTimeout: time.Millisecond * 200,
+	})
+
+	fastSrvDoneFn := testutil.HTTPServer(fastServerAddr, testutil.Options{})
+
+	monitor, err := NewMonitor(config.New(), nil, nil)
+	require.NoError(t, err)
+	require.NoError(t, monitor.Start())
+	defer monitor.Stop()
+
+	abortedRequestFn := requestGenerator(t, fmt.Sprintf("%s/ignore", slowServerAddr), emptyBody)
+	wg := sync.WaitGroup{}
+	abortedRequests := make(chan *nethttp.Request, 10000)
+	defer close(abortedRequests)
+	for i := 0; i < 10000; i++ {
+		wg.Add(1)
+		go func() {
+			defer wg.Done()
+			req := abortedRequestFn()
+			abortedRequests <- req
+		}()
+	}
+	fastReq := requestGenerator(t, fastServerAddr, emptyBody)()
+	wg.Wait()
+	slowSrvDoneFn()
+	fastSrvDoneFn()
+
+	// Ensure all captured transactions get sent to user-space
+	time.Sleep(10 * time.Millisecond)
```

Comment:
It could be more robust to use `require.Eventually` for this to succeed under more variable conditions (like gitlab).

Reply:
Thanks for the comment!
1. I copied the line from other tests in the same file, should I change it everywhere?
2. Specifically for that test `require.Eventually` won't suffice, as I'm looking for "not included", so I need for every cycle to be "not included"

Reply:
1. I would leave the other tests for another PR
2. You could do something similar with a loop and a delay for each iteration and if you find the request in any iteration that would be a fail
3. That should not be a problem here right? If any stats are not there after the first iteration, they should be available eventually on another future call to `GetHTTPStats`?

Reply:
1. Agreed
2. I can do it for sure
3. The problem is that I need to ensure a single request (fastReq) did exists

I'll add a commit with a proposed change, lmk what do you think about it

Reply:
Made the change, let me know if it looks good now

Reply:
This looks good, thanks for making the changes. There are a couple of other spots in the file with the same pattern, can we change those too also? Otherwise lgtm

Reply:
sorry for the late response, was on PTO
i'll go over it and change them as well

Reply:
added to other test as well